### PR TITLE
Forge doc  --watch implementation

### LINKED ISSUE #8484: feat(`forge doc`): add `--watch` flag
### Component

Forge

### Describe the feature you would like

allow --watch flag on forge doc to allow live updates on documentation changes

it would be good to have a live updates on doc changes when I update my contract instead of doing a build and serve every time

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by francesco-gaglione**
## Motivation

Closes #8484 

## Solution
Watch src file changes and regenerate doc

### CODE REVIEW

**Comment by francesco-gaglione** [comment]
@DaniPopes, @Evalir, @mattsse 
What do you think of this approach? Can I go deeper with this implementation or do you think we should consider something different

**Comment by DaniPopes** [comment]
Please re use the existing watch machinery which is already present for forge test and others.

Also I believe mdbook already has some sort of watch config itself, so i would prefer that if doable

**Comment by francesco-gaglione** [comment]
@DaniPopes did you mean something like the current implementation?

File: crates/forge/bin/cmd/watch.rs [hunk 1]
```diff
@@ -311,6 +313,17 @@ pub async fn watch_test(args: TestArgs) -> Result<()> {
     Ok(())
 }
 
+/// Executes a [`Watchexec`] that listens for changes in the project's src dir
```
Comment (zerosnacks) [inline crates/forge/bin/cmd/watch.rs:316]:
```suggestion
/// Executes a [`Watchexec`] that listens for changes in the project's sources directory
```

File: crates/forge/bin/cmd/doc/mod.rs [hunk 2]
```diff
@@ -62,15 +67,15 @@ pub struct DocArgs {
...
+    pub async fn run(self) -> Result<()> {
         let root = self.root.clone().unwrap_or(find_project_root_path(None)?);
         let config = load_config_with_root(Some(root.clone()));
         let project = config.project()?;
         let compiler = ProjectCompiler::new().quiet(true);
         let _output = compiler.compile(&project)?;
 
         let mut doc_config = config.doc.clone();
-        if let Some(out) = self.out {
+        if let Some(out) = self.out.clone() {
```
Comment (zerosnacks) [inline crates/forge/bin/cmd/doc/mod.rs:78]: These `clones` are likely not necessary

**Review by DaniPopes — APPROVED** [review]

**Comment by DaniPopes** [comment]
Thanks!
FYI I've edited the description to use [GitHub keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) when referencing issues.

**Pull request merged**